python/pyspark/errors/utils.py

```diff
@@ -16,9 +16,14 @@
 #
 
 import re
-from typing import Dict, Match
-
+import functools
+import inspect
+from typing import Any, Callable, Dict, Match, TypeVar, Type
 from pyspark.errors.error_classes import ERROR_CLASSES_MAP
+from py4j.java_gateway import JavaClass
```

Comment:
Can we hide this under `typing.TYPE_CHECKING`? For pure Python library package, we should avoid this whenever possible. Otherwise, the scheduled job would fail.

Reply:
Sure, just updated